### PR TITLE
Return from add_event and remove_event whether it was new.

### PR DESCRIPTION
**Pull request by afck**
## Motivation

For https://github.com/linera-io/linera-protocol/issues/1621, we will need to keep track of the oldest non-skippable event in our inbox. We will add a queue of those to the chain state. Keeping that queue up-to-date is easier if `add_event` and `remove_event` calls tell the caller whether the event is resp. was in `added_events`.

## Proposal

Return a `bool` from `add_event` resp. `remove_event` that is `true` if it is in `added_events` now (i.e. has not been removed already) resp. if it has been removed from `added_events` just now (i.e. has been added already).

In other words, if you first add and then remove, both calls will return `true`; if you first remove then add, both will return `false`. Only in the `true` case is the event a candidate for the "oldest unhandled unskippable message in the inbox".

## Test Plan

All calls in the inbox tests now assert the return value.

## Links

- https://github.com/linera-io/linera-protocol/issues/1621
- [reviewer checklist](https://github.com/linera-io/linera-protocol/blob/main/CONTRIBUTING.md#reviewer-checklist)

### CODE REVIEW

File: linera-chain/src/inbox.rs [hunk 1]
```diff
@@ -154,7 +154,9 @@ where
     }
 
     /// Consumes an event from the inbox.
-    pub(crate) async fn remove_event(&mut self, event: &Event) -> Result<(), InboxError> {
+    ///
+    /// Returns `true` if the event was already known, i.e. it was present in `added_events`.
+    pub(crate) async fn remove_event(&mut self, event: &Event) -> Result<bool, InboxError> {
```
Comment (jvff) [inline linera-chain/src/inbox.rs:159]: Optional: After looking at the tests, I'm wondering if it would make more sense to return a new type instead, something like `enum EventStatus { Queued, Processed }` (I might have gotten the variant names wrong though) :thinking:

**Review by jvff — APPROVED** [review]
Looks good! I had an idea, but I'm not sure if it makes sense or even if it does if it's worth doing now :thinking:

File: linera-chain/src/inbox.rs (hunk 1 above)
Comment (afck) [reply linera-chain/src/inbox.rs:159]: Would that be the same enum for `add_event` and `remove_event`? :thinking: 
It might be a bit ambiguous to return `Queued` from `remove_event` if the event was _removed_ from the queue.

**Review by ma2bd — APPROVED** [review]

File: linera-chain/src/inbox.rs (hunk 1 above)
Comment (jvff) [reply linera-chain/src/inbox.rs:159]: My initial thought was to use the same. But yeah, it might be hard to find the right name `{ Duplicate, FirstTime }` or something. (Not a blocker)
